src/ImageSharp/Processing/Processors/Transforms/ResamplingWeightedProcessor.Weights.cs

```diff
@@ -0,0 +1,175 @@
+            /// Applies <see cref="Vector4Extensions.Expand(float)"/> to all input vectors.
+            /// </summary>
+            /// <param name="rowSpan">The input span of vectors</param>
+            /// <returns>The weighted sum</returns>
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public Vector4 ComputeExpandedWeightedRowSum(BufferSpan<Vector4> rowSpan)
+            {
+                float* horizontalValues = this.Ptr;
+                int left = this.Left;
+                Vector4* vecPtr = (Vector4*)rowSpan.PointerAtOffset;
+                vecPtr += left;
+
+                // Destination color components
+                Vector4 result = Vector4.Zero;
+
+                for (int i = 0; i < this.Length; i++)
+                {
+                    float weight = horizontalValues[i];
+                    result += (*vecPtr).Expand() * weight;
+                    vecPtr++;
+                }
+
+                return result;
+            }
+
+            /// <summary>
+            /// Computes the sum of vectors in 'firstPassPixels' at a column pointed by 'x',
+            /// weighted by weight values, pointed by this <see cref="WeightsWindow"/> instance.
+            /// </summary>
+            /// <param name="firstPassPixels">The buffer of input vectors in row first order</param>
+            /// <param name="x">The column position</param>
+            /// <returns>The weighted sum</returns>
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public Vector4 ComputeWeightedColumnSum(PinnedImageBuffer<Vector4> firstPassPixels, int x)
+            {
+                float* verticalValues = this.Ptr;
+                int left = this.Left;
+
+                // Destination color components
+                Vector4 result = Vector4.Zero;
+
+                for (int i = 0; i < this.Length; i++)
+                {
+                    float yw = verticalValues[i];
+                    int index = left + i;
+                    result += firstPassPixels[x, index] * yw;
+                }
+
+                return result;
+            }
+        }
+
+        /// <summary>
+        /// Holds the <see cref="WeightsWindow"/> values in an optimized contigous memory region.
+        /// </summary>
+        protected class WeightsBuffer : IDisposable
+        {
+            private PinnedImageBuffer<float> dataBuffer;
+
+            /// <summary>
+            /// Initializes a new instance of the <see cref="WeightsBuffer"/> class.
+            /// </summary>
+            /// <param name="sourceSize">The size of the source window</param>
+            /// <param name="destinationSize">The size of the destination window</param>
+            public WeightsBuffer(int sourceSize, int destinationSize)
+            {
+                this.dataBuffer = new PinnedImageBuffer<float>(sourceSize, destinationSize);
```

Comment:
Any reason for not using the static method `CreateClean` to generate the clean buffer?